rust/src/notus/packages/rpm.rs

```diff
@@ -486,10 +571,52 @@ mod rpm_tests {
             version: PackageVersion("429496729542949672954294967295.0.2092".to_string()),
             release: PackageVersion("8.oe2403".to_string()),
             arch: "x86_64".to_string(),
+            module: ("".to_string(), "".to_string()),
         };
         assert!(package1 < package2);
     }
 
+    #[test]
+    pub fn test_compare_different_module() {
+        let package1 = Rpm {
+            name: "foo".to_string(),
+            epoch: 0,
+            version: PackageVersion("1.2.3".to_string()),
+            release: PackageVersion("4".to_string()),
+            arch: "x86_64".to_string(),
+            module: ("bar".to_string(), "1337".to_string()),
+        };
+        let package2 = Rpm {
+            name: "foo".to_string(),
+            epoch: 0,
+            version: PackageVersion("1.2.3".to_string()),
+            release: PackageVersion("4".to_string()),
+            arch: "x86_64".to_string(),
+            module: ("baz".to_string(), "1337".to_string()),
+        };
+        assert!(package2.partial_cmp(&package1).is_none());
+        assert!(package1.partial_cmp(&package2).is_none());
+
+        let package1 = Rpm {
+            name: "foo".to_string(),
+            epoch: 0,
+            version: PackageVersion("1.2.3".to_string()),
+            release: PackageVersion("4".to_string()),
+            arch: "x86_64".to_string(),
+            module: ("bar".to_string(), "1337".to_string()),
+        };
+        let package2 = Rpm {
+            name: "foo".to_string(),
+            epoch: 0,
+            version: PackageVersion("1.2.3".to_string()),
+            release: PackageVersion("4".to_string()),
+            arch: "x86_64".to_string(),
+            module: ("bar".to_string(), "1338".to_string()),
+        };
+        assert!(package2.partial_cmp(&package1).is_none());
+        assert!(package1.partial_cmp(&package2).is_none());
+    }
+
```

Comment:
I just tested and noticed this scan result:
```
Vulnerable package:   ruby
Installed version:    ruby-2.7.6-138.module_el8.6.0+3263+904da987.x86_64@ruby:2.7
Fixed version:      >=ruby-2.7.8-139.module_el8.8.0+3578+2b4b06da.aarch64@ruby:2.7
```

I don't think they should be comparable, because they're still different architectures (the test below is failing)
```suggestion
    
    #[test]
    pub fn test_compare_same_module_different_arch() {
        let package1 =
            Rpm::from_full_name("ruby-2.7.6-138.module_el8.6.0+3263+904da987.x86_64@ruby:2.7").unwrap();
        let package2 =
            Rpm::from_full_name("ruby-2.7.8-139.module_el8.8.0+3578+2b4b06da.aarch64@ruby:2.7").unwrap();
        assert!(package2.partial_cmp(&package1).is_none());
        assert!(package1.partial_cmp(&package2).is_none());
    }
```